### ISSUE TITLE
Wrong evaluation code in semantic segmentation metrics computing

### ISSUE THREAD

**Issue by Mooonside** [dmlc/gluon-cv#689]
The code here evaluating semantics segmentation acc:
https://github.com/dmlc/gluon-cv/blob/4c624deb53670aec337a58547f685869d00ec550/gluoncv/utils/metrics/segmentation.py#L77

The output should be either logits or softmax scores, which are of float type. Converting it to int type before doing argmax will cause a loss of precision, especially in the case of softmax scores. The same problem occurs in evaluating iou:
https://github.com/dmlc/gluon-cv/blob/4c624deb53670aec337a58547f685869d00ec550/gluoncv/utils/metrics/segmentation.py#L95

I believe the correct version should be
`predict = np.argmax(output.asnumpy(), 1).astype('int64') + 1`

**Comment by Jerryzcn** [comment]
@zhanghang1989

**Comment by bryanyzhu** [comment]
It has been fixed in PR #582 .
